### PR TITLE
[495] Setup Vagrant for development

### PR DESCRIPTION
**Pull request by riv2q**
__Story / Bug id:__ https://github.com/CodeForPoznan/volontulo/issues/495
__Reference person:__ @riv2q
__Description:__

Adding application setup option via vagrant 

__New imports / dependencies:__

* `psycopg2>=2.6`
* `pytz==2016.7 `

__What tests do I need to run to validate this change:__

* `vagrant up`
* `vagrant ssh`
* `./manage.py runserver 0.0.0.0:8000 --settings=volontulo_org.settings.dev`

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	# computers to access the VM, whereas host only networking does not.
+	config.vm.forward_port 8000, 8000
+
+	# Share an additional folder to the guest VM. The first argument is
+	# an identifier, the second is the path on the guest to mount the
+	# folder, and the third is the path on the host to the actual folder.
+	config.vm.share_folder "project", "/home/vagrant/volontulo", "."
+
+	# Enable provisioning with a shell script.
+	config.vm.provision :shell, :path => "etc/vagrant/install.sh", :args => "volontulo"
```
Comment (MarekBleschke) [inline Vagrantfile]: I think 'volontulo' arg isn't used anywhere in install.sh

File: etc/vagrant/install.sh [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    # Create vagrant pgsql superuser
+    su - postgres -c "createuser -s vagrant"
+fi
+
+# virtualenv global setup
+if ! command -v pip; then
+    easy_install -U pip
+fi
+if [[ ! -f /usr/local/bin/virtualenv ]]; then
+    pip3.4 install virtualenv virtualenvwrapper stevedore virtualenv-clone
```
Comment (MarekBleschke) [inline etc/vagrant/install.sh]: Why do we need to install setevedor and virtualenv-clone?

File: etc/vagrant/install.sh [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+# virtualenv setup for project
+su - vagrant -c "/usr/local/bin/virtualenv $VIRTUALENV_DIR --python=/usr/bin/python3.4 --no-site-packages && \
+    echo $PROJECT_DIR > $VIRTUALENV_DIR/.project && \
+    $VIRTUALENV_DIR/bin/pip install -r $PROJECT_DIR/requirements/dev.txt"
+
+
+cp $PROJECT_DIR/etc/local_config.yaml.sample $PROJECT_DIR/local_config.yaml
+sed '/^secret_key/ d' $PROJECT_DIR/local_config.yaml > $PROJECT_DIR/tmp.yaml && mv $PROJECT_DIR/tmp.yaml $PROJECT_DIR/local_config.yaml
+UUID=a63eb5ef-3b25-4595-846a-5d97d99486f0
```
Comment (MarekBleschke) [inline etc/vagrant/install.sh]: Maybe this could be generated as in install instructions?

File: apps/volontulo/package.json [hunk 4]
```diff
@@ -1,13 +1,21 @@
...
   },
   "devDependencies": {
+    "core-util-is": "~1.0.2",
     "gulp": "^3.6.0",
+    "gulp-iconify": "~0.4.0",
+    "gulp-sass": "^2.3.2",
     "gulp-util": "^3.0.6",
-    "gulp-sass": "^2.0.4",
-    "gulp-iconify": "~0.4.0"
+    "isarray": "~2.0.1",
```
Comment (MarekBleschke) [inline apps/volontulo/package.json]: A lot of new dependencies, are they used anywhere in project?

Comment (riv2q) [inline apps/volontulo/package.json]: those are dependencies for npm to install properly without errors

File: etc/vagrant/install.sh (hunk 2 above)
Comment (riv2q) [inline etc/vagrant/install.sh]: no reason just for fun, those are bind with vagrant ssh command and start by default

File: etc/vagrant/install.sh (hunk 3 above)
Comment (riv2q) [inline etc/vagrant/install.sh]: workflow was copied from Manual Project Set Up in readme

File: Vagrantfile (hunk 1 above)
Comment (riv2q) [inline Vagrantfile]: ok, good point, check again 👍

**Comment by magul** [comment]
Travis's tests are failing here.

**Comment by magul** [comment]
@riv2q 

You've changed dev's settings, that they use Postgres. Please, create separate settings file for Vagrant.
See:
```
1.01s$ coverage run --source='./apps' manage.py test --settings=volontulo_org.settings.dev -v 3
nosetests -v --verbosity=3
nose.config: INFO: Ignoring files matching ['^\\.', '^_', '^setup\\.py$']
Creating test database for alias 'default' ('test_volontulo')...
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.4.2/lib/python3.4/site-packages/django/db/backends/base/base.py", line 130, in ensure_connection
    self.connect()
  File "/home/travis/virtualenv/python3.4.2/lib/python3.4/site-packages/django/db/backends/base/base.py", line 119, in connect
    self.connection = self.get_new_connection(conn_params)
  File "/home/travis/virtualenv/python3.4.2/lib/python3.4/site-packages/django/db/backends/postgresql_psycopg2/base.py", line 176, in get_new_connection
    connection = Database.connect(**conn_params)
  File "/home/travis/virtualenv/python3.4.2/lib/python3.4/site-packages/psycopg2/__init__.py", line 164, in connect
    conn = _connect(dsn, connection_factory=connection_factory, async=async)
psycopg2.OperationalError: FATAL:  role "volontulo" does not exist
```

**Comment by riv2q** [comment]
will do

File: Vagrantfile [hunk 5]
```diff
@@ -0,0 +1,34 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+Vagrant::Config.run do |config|
+	# Base box to build off, and download URL for when it doesn't exist on the user's system already
+	config.vm.box = "ubuntu/trusty32"
```
Comment (magul) [inline Vagrantfile]: docker use 16.04. I believe we should use consistend image here.

It looks that newest ubuntu images are stored on Atlas: https://atlas.hashicorp.com/ubuntu

File: etc/vagrant/bashrc [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+# enable programmable completion features (you don't need to enable
+# this, if it's already enabled in /etc/bash.bashrc and /etc/profile
+# sources /etc/bash.bashrc).
+if [ -f /etc/bash_completion ] && ! shopt -oq posix; then
+    . /etc/bash_completion
+fi
+
+export WORKON_HOME=$HOME/.virtualenvs
+export VIRTUALENVWRAPPER_PYTHON=/usr/bin/python3.4
+source /usr/local/bin/virtualenvwrapper.sh
```
Comment (magul) [inline etc/vagrant/bashrc]: I would suggest to remove virtualenv (and wrapper), as we are already virtualizing environment here.

File: etc/vagrant/bashrc [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+# enable programmable completion features (you don't need to enable
+# this, if it's already enabled in /etc/bash.bashrc and /etc/profile
+# sources /etc/bash.bashrc).
+if [ -f /etc/bash_completion ] && ! shopt -oq posix; then
+    . /etc/bash_completion
+fi
+
+export WORKON_HOME=$HOME/.virtualenvs
+export VIRTUALENVWRAPPER_PYTHON=/usr/bin/python3.4
```
Comment (magul) [inline etc/vagrant/bashrc]: we should use newest python available

File: etc/vagrant/etc-bash.bashrc [hunk 8]
```diff
@@ -0,0 +1,68 @@
+# System-wide .bashrc file for interactive bash(1) shells.
```
Comment (magul) [inline etc/vagrant/etc-bash.bashrc]: what's the point to use two `.bashrc` file (one user-based and one system-wide)?

Comment (magul) [inline etc/vagrant/etc-bash.bashrc]: actually I don't fully understand the reason to store `.bashrc` files here at all. What's the reason, that we cannot use standard `.bashrc` provided by Vagrant box?

File: Vagrantfile (hunk 1 above)
Comment (magul) [inline Vagrantfile]: It looks that is used as `$PROJECT_NAME` in `install.sh`

File: etc/vagrant/install.sh [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+PROJECT_DIR=/home/vagrant/$PROJECT_NAME
+VIRTUALENV_DIR=/home/vagrant/.virtualenvs/$PROJECT_NAME
+
+
+# Install essential packages from Apt
+apt-get update -y
+# Python dev packages
+apt-get install -y build-essential python python3-dev npm
+# python-setuptools being installed manually
+wget https://bootstrap.pypa.io/ez_setup.py -O - | python3.4
```
Comment (magul) [inline etc/vagrant/install.sh]: I just use `pip3 install --upgrade pip` on docker, could we figure out what's a better solution?

File: etc/vagrant/install.sh [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+# Dependencies for image processing with Pillow (drop-in replacement for PIL)
+# supporting: jpeg, tiff, png, freetype, littlecms
+# (pip install pillow to get pillow itself, it is not in requirements.txt)
+apt-get build-dep -y python-imaging
+apt-get install -y nodejs libjpeg-dev libtiff-dev zlib1g-dev libfreetype6-dev liblcms2-dev libjpeg8 libjpeg62-dev
+# Git (we'd rather avoid people keeping credentials for git commits in the repo, but sometimes we need it for pip requirements that aren't in PyPI)
```
Comment (magul) [inline etc/vagrant/install.sh]: what kind of requirements? I don't recall any such kind.

File: etc/vagrant/install.sh [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+
+# Django project setup
+cp $PROJECT_DIR/etc/vagrant/settings.py $PROJECT_DIR/volontulo_org/settings/dev_vagrant.py
+su - vagrant -c "source $VIRTUALENV_DIR/bin/activate && cd $PROJECT_DIR && \
+   ./manage.py migrate --settings=volontulo_org.settings.dev_vagrant && \
+   ./manage.py loaddata initial/data.json --settings=volontulo_org.settings.dev_vagrant"
+
+
+# Reinstall Node
+su - -c "curl -sL https://deb.nodesource.com/setup_4.x | sudo -E bash -"
```
Comment (magul) [inline etc/vagrant/install.sh]: Use NVM instead https://github.com/creationix/nvm#install-script

**Pull request closed without merging**